spring-integration-core/src/main/java/org/springframework/integration/support/MutableMessageBuilder.java

```diff
@@ -73,6 +78,10 @@ public T getPayload() {
 	 */
 	public static <T> MutableMessageBuilder<T> fromMessage(Message<T> message) {
 		Assert.notNull(message, "message must not be null");
+		if (message instanceof CloneableMessage) {
+			logger.warn("For CloneableMessages consider to use regular MessageBuilder. " +
+					"Otherwise custom Message context may be lost.");
+		}
```

Comment:
OK. I'll like the current abstract solution better :smile: 

But I want to understand what's wrong with it.
Thanks